spec/prog/minio/setup_minio_spec.rb

```diff
@@ -68,20 +68,32 @@
   end
 
   describe ".configure_minio" do
-    it "configures minio" do
+    it "configures minio if not started" do
       expect(nx.minio_server.cluster.servers.first).to receive(:private_ipv4_address).and_return("192.168.0.0")
-      expect(nx.minio_server.vm.sshable).to receive(:cmd).with("sudo tee -a /etc/default/minio", stdin: <<SH)
-MINIO_VOLUMES="/minio/dat1"
-MINIO_OPTS="--console-address :9001"
-MINIO_ROOT_USER="minio-admin"
-MINIO_ROOT_PASSWORD="dummy-password"
-SH
-      expect(nx.minio_server.vm.sshable).to receive(:cmd).with("sudo tee -a /etc/hosts", stdin: <<SH)
-192.168.0.0 minio-cluster-name0.minio.ubicloud.com
-SH
-      expect(nx.minio_server.vm.sshable).to receive(:cmd).with("sudo chown -R minio-user:minio-user /etc/default/minio")
+      expect(nx.minio_server.vm.sshable).to receive(:cmd).with("common/bin/daemonizer --check configure_minio").and_return("NotStarted")
+      expect(nx.minio_server.vm.sshable).to receive(:cmd).with("common/bin/daemonizer 'sudo minio/bin/configure-minio' configure_minio", stdin: "{\"minio_config\":\"MINIO_VOLUMES=\\\"/minio/dat1\\\"\\nMINIO_OPTS=\\\"--console-address :9001\\\"\\nMINIO_ROOT_USER=\\\"minio-admin\\\"\\nMINIO_ROOT_PASSWORD=\\\"dummy-password\\\"\\n\",\"hosts\":\"::1 ip6-localhost ip6-loopback\\nfe00::0 ip6-localnet\\nff00::0 ip6-mcastprefix\\nff02::1 ip6-allnodes\\nff02::2 ip6-allrouters\\nff02::3 ip6-allhosts\\n192.168.0.0 minio-cluster-name0.minio.ubicloud.com\\n\"}")
```

Comment:
Isn't multiline stdin more readable and maintainable?

Reply:
Do you like it now?